core/src/main/scala/scalaz/StrictTree.scala

```diff
@@ -48,10 +48,21 @@ case class StrictTree[A](
 
   /** Maps the elements of the StrictTree into a Monoid and folds the resulting StrictTree. */
   def foldMap[B: Monoid](f: A => B): B =
-    runBottomUp(foldMapReducer(f))
+    foldLeft(Monoid[B].zero)((a, b) => Monoid[B].append(b, f(a)))
+
+  def foldLeft[B](z: B)(f: (A, B) => B): B = {
+    var stack = this :: Nil
+    var result = z
+    while (stack.nonEmpty) {
+      val head :: tail = stack
+      result = f(head.rootLabel, result)
+      stack = head.subForest ::: tail
```

Comment:
hmm. this could avoid a `:::` by having `stack` be a `List[List[A]]` perhaps

```scala
var stack = List(this) :: Nil
var result = z
while (stack.nonEmpty) {
  val (head :: tail) :: rest = stack
  result = f(head.rootLabel, result)
  stack = (head.subForest :: tail :: stack) dropWhile (_.isEmpty)
}
result
```

or something like that (but tested)